cmd/account/reset.go

```diff
@@ -75,6 +87,23 @@ func (o *resetOptions) complete(cmd *cobra.Command, args []string) error {
 	if len(args) != 1 {
 		return cmdutil.UsageErrorf(cmd, "The name of Account CR is required for reset command")
 	}
+
+	if o.editState {
+		if o.stateInput == "" {
+			return cmdutil.UsageErrorf(cmd, "You must provide a state to put the account into. Supported states are Ready or Failed")
+		}
+		stateIn := strings.ToLower(o.stateInput)
+
+		switch stateIn {
+		case "ready":
+			o.setState = stateReady
+		case "failed":
+			o.setState = stateFailed
+		default:
+			return cmdutil.UsageErrorf(cmd, "Unsupported state.  Supported states are Ready or Failed")
+		}
+	}
```

Comment:
Why do we need to check this twice? Can't we move this to the run function so the `if o.editState` is only done once? Translating the string to state could be a function called from there.